explainers/anti-tracking.md

```diff
@@ -0,0 +1,247 @@
+
+<!-- TOC depthTo:3 -->
+
+- [Blocking credentials](#blocking-credentials)
+  - [Possible spellings](#possible-spellings)
+  - [Tentatively-rejected spellings](#tentatively-rejected-spellings)
+- [Blocking normal link decoration](#blocking-normal-link-decoration)
+- [Blocking privileged link decoration](#blocking-privileged-link-decoration)
+- [Privileged server-side collaboration](#privileged-server-side-collaboration)
+- [Appendices](#appendices)
+  - [Security concerns](#security-concerns)
+  - [Behavior on an unexpected response](#behavior-on-an-unexpected-response)
+
+<!-- /TOC -->
+
+This explainer attempts to solve the problems identified in issues
+[#422](https://github.com/WICG/webpackage/issues/422) and
+[#423](https://github.com/WICG/webpackage/issues/423). Specifically, we're
+worried that a distributor might be able to use the fact that it can choose
+which exchange or package, signed by a publisher, to serve in response to a
+client's request, to transfer its notion of the user's identity to the
+publisher.
+
+The approach this document takes is to prevent the distributor from accessing
+its notion of the user's identity. There are several routes by which a
+distributor could learn the user's identity:
+
+1. The browser could send the distributor its top-level credentials. [Countermeasures](#blocking-credentials)
+1. The source of a link could use the same link-decoration tools as it might use
+   in a cross-origin link, which other efforts are blocking. Note that the
+   source of a link and the distributor of a package could be the same origin.
+   [Countermeasures](#blocking-normal-link-decoration)
+1. The source of a link could decorate links to its distributor to send a user
+   ID in ways an unrelated cross-origin target couldn't be expected to receive.
+   [Countermeasures](#blocking-privileged-link-decoration)
+1. The source of a link could collaborate with its distributor on the server
+   side to send a user ID in ways an unrelated cross-origin target couldn't be
+   induced to collaborate.
+   [Discussion](#privileged-server-side-collaboration)
+
+These are tackled individually in the following sections.
+
+## Blocking credentials
+
+The navigation to a signed package (or exchange) must be done without
+credentials. This is roughly [credentials
+mode](https://fetch.spec.whatwg.org/#concept-request-credentials-mode) ==
+`"omit"`, with changes to Fetch to make it work for navigations.
+
+There are [security concerns](#security-concerns) for allowing an attacker to
+load an arbitrary site without credentials, so if the target of an
+uncredentialed navigation isn't a signed package, the navigation should
+[fail](#behavior-on-an-unexpected-response).
+
+There are lots of possible spellings for a navigation that omits credentials:
+
+### Possible spellings
+
+<a id="credentials-omit"></a>
+
+#### An attribute to set [credentials-mode](https://fetch.spec.whatwg.org/#concept-request-credentials-mode) to "omit"
```

Comment:
Something that might be worth writing out explicitly:

If I'm understanding things right, this option and the second option are qualitatively different from the remaining ones and prenavigate. These say that there are two ways a referrer may choose to enter a first-party site, credentialed or uncredentialed. The other options don't change the semantics of top-level navigations (aside from the SXG origin substitution capability itself), but are trying to slot an uncredentialed package fetch somehow.

Reply:
I've written this.